### DISCUSSION TITLE
changing focus in table based on selected row

### DISCUSSION THREAD

**Discussion by Masaomix** [zauberzeug/nicegui#2853] [Q&A] [answered]
### Question

With the help of Falco i manged to highlight a certain row in a table, based on it's value.  In my app this value changes automatically when the next song is being played (i'ts kind of a media player).  The table in my app is sometimes very big (thousends of rows). So, the song which is played at the moment is highlighted, but i would have to scroll and turn the pages to actually see the highlighted song. I need a way, based on the value of the row to focus on this row.

Until now i had two Ideas:

Keeping my pages small (15 rows per page or so). In this case i would only have go to the page which has the active song. 
This might be the easier solution, but i don't have an idea how to turn to a certain page based on row / cell value.

Using ui.scroll_area and scroll_to.  At the moment my rows are set to 300 per page.  It would be nice to directly focus on the active song, but I would stilll need to go to the page which contains the song. I experimented with this, but again, i have no idea how to use scrolling based on the value of a row. 

This is the probably the last big puzzle i have to face with my app, so any help would be very appreciated. Here is a very simplified code example:

```py
from nicegui import app, ui
from nicegui.events import KeyEventArguments

columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left', 'sortable': True},
]
rows = [
    {'name': 'Song1'},
    {'name': 'Song2'},
    {'name': 'Song3'},
    {'name': 'Song4'},
    {'name': "Song'5"},
    {'name': 'Song6'},
    {'name': 'Song7'},
    {'name': 'Song8'},
    {'name': 'Song9'},
]

song_played = "Song'5"
search_index = next((i for i, row in enumerate(rows) if row['name'] == song_played), None)
table1 = ui.table(columns=columns, rows=rows, row_key='name', pagination={'rowsPerPage': 3})

table1.add_slot('body-cell-name', '''
    <q-td key="name" :props="props">
        <q-badge :color="props.rowIndex === ''' + str(search_index) + ''' ? 'red' : 'black'" multiline>
            {{ props.value }}
        </q-badge>
    </q-td>
''')
ui.run(show= False, reload= False)
```

**Comment by Masaomix** [comment]
Hmmm, somehow the add_slot part is not shown correctly here... this should be between the triple quotation marks:

    <q-td key="name" :props="props">
        <q-badge :color="props.rowIndex === ''' + str(search_index) + ''' ? 'red' : 'black'" multiline>
            {{ props.value }}
        </q-badge>
    </q-td>

**Comment by falkoschindler** [comment, marked as answer]
Hi @Masaomix,

You can specify the page index as part of the `pagination` object:

```py
ROWS_PER_PAGE = 3
page = search_index // ROWS_PER_PAGE + 1  # 1-based index
table1 = ui.table(..., pagination={'rowsPerPage': ROWS_PER_PAGE, 'page': page})
```
